src/test/kotlin/org/eclipse/hara/ddiclient/integrationtest/DdiClientHttpRequestsTest.kt

```diff
@@ -0,0 +1,505 @@
+            `test #5-2= next request should use target token, when target token is enabled in server`()
+            `test #5-3= next request should use gateway token, when target token is disabled in server`()
+            `test #5-4= next request should keep using gateway token, when target token is enabled in server`()
+            `test #5-5= next request should switch to target token, when gateway token is disabled in server`()
+            `test #5-6= next request should keep using target token, when gateway token is enabled in server`()
+        }
+    }
+
+    private suspend fun `test #5-1= request should fail, when both tokens are disabled in server`() {
+        logCurrentFunctionName()
+        expectPollingOnlyMessage()
+        expectedServerResponses.apply {
+            add(targetTokenMessage(HttpURLConnection.HTTP_UNAUTHORIZED))
+            add(gatewayTokenMessage(HttpURLConnection.HTTP_UNAUTHORIZED))
+        }
+        startSubTestTest()
+    }
+
+    private suspend fun `test #5-2= next request should use target token, when target token is enabled in server`() {
+        logCurrentFunctionName()
+        enableTargetTokenInServer()
+        expectPollingAndIdleMessages()
+        expectedServerResponses.apply {
+            add(targetTokenMessage(HttpURLConnection.HTTP_OK))
+        }
+        startSubTestTest()
+    }
+
+    private suspend fun `test #5-3= next request should use gateway token, when target token is disabled in server`() {
+        logCurrentFunctionName()
+        enableTargetTokenInServer(false)
+        enableGatewayTokenInServer(true)
+        expectPollingOnlyMessage()
+        expectedServerResponses.apply {
+            add(targetTokenMessage(HttpURLConnection.HTTP_UNAUTHORIZED))
+            add(gatewayTokenMessage(HttpURLConnection.HTTP_NOT_MODIFIED))
+        }
+        startSubTestTest()
+    }
+
+    private suspend fun `test #5-4= next request should keep using gateway token, when target token is enabled in server`() {
+        logCurrentFunctionName()
+        enableTargetTokenInServer(true)
+        expectPollingOnlyMessage()
+        expectedServerResponses.apply {
+            add(gatewayTokenMessage(HttpURLConnection.HTTP_NOT_MODIFIED))
+        }
+        startSubTestTest()
+    }
+
+    private suspend fun `test #5-5= next request should switch to target token, when gateway token is disabled in server`() {
+        logCurrentFunctionName()
+        enableGatewayTokenInServer(false)
+        expectPollingOnlyMessage()
+        expectedServerResponses.apply {
+            add(gatewayTokenMessage(HttpURLConnection.HTTP_UNAUTHORIZED))
+            add(targetTokenMessage(HttpURLConnection.HTTP_NOT_MODIFIED))
+        }
+        startSubTestTest()
+    }
+
+    private suspend fun `test #5-6= next request should keep using target token, when gateway token is enabled in server`() {
+        logCurrentFunctionName()
+        enableGatewayTokenInServer(false)
```

Comment:
@SaeedRe I guess this should have been:
```suggestion
        enableGatewayTokenInServer(true)
```